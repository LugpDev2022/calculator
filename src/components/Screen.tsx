import { CalcState } from '../calculatorReducer';
import ResultDisplay from './ResultDisplay';

interface Props {
  calcState: CalcState;
}

const Screen: React.FC<Props> = ({ calcState }) => {
  const { operation, result, error } = calcState;

  return (
    <div
      className='
        screen-container
        my-5
        overflow-hidden
        px-5
        py-2
        rounded-2xl
        w-full
      '
    >
      <span
        className='
          operation
          block
          font-semibold
          min-h-[20px]
          text-lg
          text-right
          w-full
          sm:min-h-[28px]
          transition
        '
      >
        {operation ? operation : '0'}
      </span>
      <ResultDisplay error={error} result={result} />
    </div>
  );
};

export default Screen;
