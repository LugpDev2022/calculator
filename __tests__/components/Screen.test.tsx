import React from 'react';

import { vi } from 'vitest';
import { render, screen } from '@testing-library/react';

import Screen from '../../src/components/Screen';
import {
  CalcStateContext,
  CalcStateContextType,
} from '../../src/context/CalcStateContext';

vi.mock('../../src/components/ResultDisplay', () => ({
  default: () => <span>5</span>,
}));

describe('Tests on <Screen />', () => {
  const contextValue: CalcStateContextType = {
    dispatch: vi.fn(),
    error: '',
    operation: '2+2',
    result: 10,
  };

  test('should match snapshot ', () => {
    const { container } = render(
      <CalcStateContext.Provider value={contextValue}>
        <Screen />
      </CalcStateContext.Provider>
    );

    expect(container).toMatchSnapshot();
  });

  test('should show the operation', () => {
    render(
      <CalcStateContext.Provider value={contextValue}>
        <Screen />
      </CalcStateContext.Provider>
    );

    expect(screen.queryByText(contextValue.operation)).toBeTruthy();
  });

  test('should show 0 when operation not written', () => {
    render(
      <CalcStateContext.Provider value={{ ...contextValue, operation: '' }}>
        <Screen />
      </CalcStateContext.Provider>
    );

    expect(screen.queryByText(0)).toBeTruthy();
  });
});
