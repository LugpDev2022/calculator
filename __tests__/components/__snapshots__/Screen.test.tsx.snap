// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests on <Screen /> > should match snapshot  1`] = `
<div>
  <div
    class=" screen-container my-5 overflow-hidden px-5 py-2 rounded-2xl w-full "
  >
    <span
      class=" operation block font-semibold min-h-[20px] text-lg text-right w-full sm:min-h-[28px] transition "
    >
      2+2
    </span>
    <span>
      5
    </span>
  </div>
</div>
`;
